library/network_connections.py

```diff
@@ -1789,6 +1845,18 @@ def run_prepare(self):
             connection["nm.exists"] = exists
             connection["nm.uuid"] = uuid
 
+    def start_transaction(self):
+        Cmd.start_transaction(self)
+        self._checkpoint = self.nmutil.create_checkpoint()
+
+    def rollback_transaction(self, idx, action, error):
+        Cmd.rollback_transaction(self, idx, action, error)
+        self.nmutil.rollback_checkpoint(self._checkpoint)
+
+    def finish_transaction(self):
+        Cmd.finish_transaction(self)
+        self.nmutil.destroy_checkpoint(self._checkpoint)
```

Comment:
both `destroy_checkpoint()` and `rollback_checkpoint()` do nothing if `self._checkpoint` is `None` (good).

But both calls will invalidate the checkpoint path. It would be nice if the `self._checkpoint` attribute would be reset to None, and not leave this invalid state there. Since both functions can raise an exception, to get it really right, this should be
```python
     path = self._checkpoint
     self._checkpoint = None
     self.nmutil.destroy_checkpoint(path)
```
which is kinda cumbersome...

Reply:
I am not sure that  discarding the checkpoint before actually destroying it is right. Does an exception always indicate that the checkpoint is invalid? I changed to to at least resetting the checkpoint to None after destroying it.

Reply:
Arguably, it doesn't really matter because on every execption from this point the program will terminated (and fail). But there really is no point in calling `destroy_checkpoint()` or `rollback_checkpoint()` more than once. After calling it, the path is certainly no longer suitable to destroy/rollback again and clearing the field seems more correct to me.

Reply:
I put resetting the checkpoint into a finally block, IMHO it makes it more clear why it is there.